### PR TITLE
(Epic Web) Adds more structured data to workshops

### PR DESCRIPTION
**Pull request by Creeland**
## Problem

Pricing information is not being shown in our search results. 

Workshops are also not being considered "courses" by Google. So they aren't being displayed in the courses carousel.

image
image

## Solution

Add additional structured data to the `CourseJsonLd` components in workshop-template and buy

molecule

### CODE REVIEW

File: apps/epic-web/src/templates/workshop-template.tsx [hunk 1]
```diff
@@ -432,21 +432,34 @@ const Header: React.FC<{module: Module; canView: boolean}> = ({
...
-}: {
-  title: string
-  description?: string | null | undefined
-}) => (
+const CourseMeta = ({product}: {product: SanityProduct}) => (
   <CourseJsonLd
-    courseName={title}
-    description={description || ''}
+    courseName={product.title || ''}
+    description={product.description || ''}
```
Comment (zacjones93) [inline apps/epic-web/src/templates/workshop-template.tsx:438]: we should beef these up. Crawlers expect certain char lengths for title and description, this goes for where ever normal SEO is being set as well (could be a different PR?) 

image

File: apps/epic-web/src/pages/buy.tsx [hunk 2]
```diff
@@ -86,6 +89,55 @@ const BuyPage: React.FC<
...
+        <CourseJsonLd
+          courseName="Full Stack Vol 1"
+          description="The most comprehensive guide to professional web development by Kent C. Dodds."
+          provider={{
+            name: 'Kent C. Dodds',
+            type: 'Person',
+            url: isBrowser()
+              ? document.location.href
+              : process.env.NEXT_PUBLIC_URL,
+          }}
```
Comment (zacjones93) [inline apps/epic-web/src/pages/buy.tsx:101]: `sameAs` is being passed by default which we can add kents socials here and personal site. This could be 

```json
{
  sameAs: ['https://kentcdodds.com', 'https://twitter.com/kentcdodds', 'https://github.com/kentcdodds'] 
}
```

https://schema.org/sameAs

image

Comment (Creeland) [reply apps/epic-web/src/pages/buy.tsx:101]: Unfortunately, the provider prop doesn't support sameAs. I think that's the intended use of the URL prop. But, you can't pass an array to it. 

I'll set Kent's personal site as that URL.

**Review by zacjones93 — APPROVED** [review]
looks great
